src/backend/commands/matview.c

```diff
@@ -468,6 +468,34 @@ ExecRefreshMatView(RefreshMatViewStmt *stmt, const char *queryString,
 	 */
 	CheckTableNotInUse(matviewRel, "REFRESH MATERIALIZED VIEW");
 
+	/*
+	 * Fast path to REFRESH a view:
+	 * avoid do the real REFRESH if the data of view
+	 * is up to date. The data should be the logically same as after
+	 * REFRESH when there is data changed since latest REFRESH.
+	 * In that case we may save a lot, ex: a cron task REFRESH view periodically
+	 * or manually executed by users each time.
+	 *
+	 * Set this feature default to true, but let uesrs decide if they intend
+	 * to do a real REFRESH.
+	 */
+	if (gp_enable_refresh_fast_path &&
```

Comment:
also !oldPopulated need to refresh.

Reply:
It's included in `MatviewIsGeneralyUpToDate()`

Reply:
OK ,better to add a test to cover this case.

Reply:
The feature is on by default, so there are many cases have tested no-populated mvs, example:
https://github.com/cloudberrydb/cloudberrydb/blob/74808061fa642568f714f32743f3b64fe6ccd3a2/src/test/regress/expected/incremental_matview.out#L15

```sql
CREATE INCREMENTAL MATERIALIZED VIEW mv_ivm_1 AS SELECT i,j,k FROM mv_base_a a INNER JOIN mv_base_b b USING(i) WITH NO DATA;
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'i' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT * FROM mv_ivm_1 ORDER BY 1,2,3;
ERROR:  materialized view "mv_ivm_1" has not been populated
HINT:  Use the REFRESH MATERIALIZED VIEW command.
REFRESH MATERIALIZED VIEW mv_ivm_1;
SELECT * FROM mv_ivm_1 ORDER BY 1,2,3;
 i | j  |  k  
---+----+-----
 1 | 10 | 101
 2 | 20 | 102
 3 | 30 | 103
 4 | 40 | 104
(4 rows)
```